src/patch.dev.js

```diff
@@ -77,26 +77,43 @@ if (React.createElement.isPatchedByReactHotLoader) {
 // https://github.com/gaearon/react-hot-loader/issues/249
 function isReactRouterish(type) {
   return type && (
-    type.displayName === 'Route' ||
-    type.name === 'Route' || // In case Ryan and Michael embrace ES6 classes
-    type.displayName === 'IndexRoute' ||
-    type.name === 'IndexRoute'
+    type.displayName === 'Router' ||
+    type.name === 'Router' // In case Ryan and Michael embrace ES6 classes
   );
 }
 
+function forceUpdateComponentOfRouteAndChildRoutes(route) {
+  // TODO: check whether it is possible to also handle the `getComponent` case here
+  const components = []
+    .concat(route.component, route.components && Object.values(route.components))
```

Comment:
`Object.values` may not work in older browsers, we shoot for wide compatibility.